web/web.go

```diff
@@ -240,9 +241,11 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
-	if err != nil {
-		panic(err)
+	if hostname == "" {
+		hostname, err = os.Hostname()
+		if err != nil {
+			panic(err)
+		}
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
```

Comment:
I'd suggest removing that behaviour actually, as that may ultimately cause confusion if I asked to scrape localhost but that's not what's showing up.